source-stripe-native/source_stripe_native/api.py

```diff
@@ -0,0 +1,495 @@
+            parent_data = doc
+            id = parent_data.id
+            child_data = _capture_substreams(
+                            cls_child,
+                            search_name,
+                            id,
+                            parent_data,
+                            http,
+                            log
+                        )
+
+            if child_data is None:
+                pass # move to next customer
+            async for doc in child_data:
+                doc.parent_id = parent_data.id
+                doc.meta_ = cls_child.Meta(op="u")
+                yield doc 
+
+    if result.has_more:
+        yield result.data[-1].id
+    else:
+        return
+
+async def _capture_substreams(
+    cls_child,
+    search_name,
+    id,
+    parent_data,
+    http,
+    log
+):
+    """_capture_substreams works by handling the child_stream query and pagination.
+    it requires the parent stream data, along with the item id and the parent search name.
+    """
+
+    child_url = f"{API}/{search_name}/{id}/{cls_child.SEARCH_NAME}"
+    parameters = cls_child.PARAMETERS
+    if "subscription" in parameters.keys():
+        parameters["subscription"] = id
+        child_url = f"{API}/{cls_child.SEARCH_NAME}"
+    elif "setup_intent" in parameters.keys():
+        parameters["setup_intent"] = id
+        child_url = f"{API}/{cls_child.SEARCH_NAME}"
+    elif cls_child.NAME == "UsageRecords":
+        child_url = f"{API}/subscription_items/{id}/{cls_child.SEARCH_NAME}"
+
+    while True:
+        if cls_child.NAME == "Persons" and parent_data.controller["requirement_collection"] == "stripe" :
+            break
+        result_child = ListResult[cls_child].model_validate_json(
+        await http.request(log, child_url, method="GET", params=parameters)
+    )
+
+        for doc in result_child.data:
+            yield doc
+
+        if result_child.has_more is True:
+            parameters["starting_after"] = result_child.data[-1].id
+        elif result_child.has_more is False:
+            break
+
+
+async def fetch_incremental_no_events(
+    cls,
+    http: HTTPSession,
+    log: Logger,
+    log_cursor: LogCursor,
+) -> AsyncGenerator:
+    """ Note: Stripe's data is always served in reverse-chronological order.
+    fetch_incremental_no_events works very similar to fetch_backfilling. This method
+    handles streams that do not have valid Event Types.
+    It works by calling each individual stream endpoint and parsing each result,
+    created after the last log_cursor, with its model.
+    """
+
+    stop = True
+
+    url = f"{API}/{cls.SEARCH_NAME}"
+    parameters = {"limit": 100}
+    recent = []
+
+
+    _cls: Any = cls  # Silence mypy false-positive
+
+    while stop:
+        result = ListResult.model_validate_json(
+        await http.request(log, url, method="GET", params=parameters)
+    )
+        for results in result.data:
+            if _s_to_dt(results["created"]) > log_cursor:
```

Comment:
It looks like the modeling is not right here.  `ListResult` has `data: List[Item]` but `Item = TypeVar("Item")`. The modeling should updated so we don't have to access the "created" property of an untyped variable like this.

Reply:
Updated here 00467212d8135d9c433b5806262295c55c368f4c